src/math.rs

```diff
@@ -186,24 +186,20 @@ pub fn crt(r: &[i64], m: &[i64]) -> (i64, i64) {
 /// assert_eq!(math::floor_sum(6, 5, 4, 3), 13);
 /// ```
 pub fn floor_sum(n: i64, m: i64, mut a: i64, mut b: i64) -> i64 {
+    assert!(0 <= n && n < 1i64 << 32);
+    assert!(1 <= m && m < 1i64 << 32);
     let mut ans = 0;
```

Comment:
Since overflows occur in C++ without special handling, they use `unsigned long long` here to extend the range of values returned correctly; implementing this in Rust would require a lot of casting (`as u64`) and functions like `u64::overflowing_sub`. Since this is cumbersome, I've written the internal computation in i64, but this may have led to more cases of overflowing only in the middle of the computation. Which would be better?

Reply:
Why not using `wraping_*` variants?